h/services/auth_token.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+import jwt
+
+from h import models
+from h.auth.tokens import LegacyClientJWT, Token
+
+
+class AuthTokenService(object):
+    def __init__(self, session, client_secret):
+        self._session = session
+        self._client_secret = client_secret
+
+        self._validate_cache = {}
+
+    def validate(self, token_str):
+        """
+        Load and validate a token.
+
+        This will return a token object implementing
+        ``h.auth.interfaces.IAuthenticationToken``, or ``None`` when the token
+        cannot be found, is not a legacy JWT token, or is not valid.
+
+        :param token_str: the token string
+        :type token_str: unicode
+
+        :returns: the token object, if found and valid, or ``None``.
+        """
+
+        if token_str in self._validate_cache:
```

Comment:
Hmm ... are there any scenarios in which this cache needs to be explicitly invalidated?